design/one-pager-mrds.md

```diff
@@ -0,0 +1,248 @@
+
+MRs can automatically write sensitive connection details like addresses,
+usernames, and passwords to a Kubernetes Secret. We call these connection
+details. Today it's very hard to discover what connection details an MR
+supports. The only way to do so is to create an MR and see what it writes to its
+connection secret. This is tracked in [issue 1143][1] - Crossplane's third most
+upvoted issue.
+
+Crossplane bundles related MRs together into a provider - like provider-github
+or provider-aws-ec2. Today some of these providers install up to 100 MRs. With
+v2 this'll double, because v2 introduces namespaced MRs but keeps the existing
+cluster scoped MRs for backward compatibility.
+
+Each MR is powered by a CustomResourceDefinition (CRD) and a controller. Each
+CRD has a performance penalty on the Kubernetes API server - mostly memory
+usage. Each CRD also creates one or more API endpoints that clients like kubectl
+may need to walk.
+
+It's not possible to install only some of a provider's MRs. It's all or nothing.
+The ability to install only the MRs you intend to use is tracked in [issue
+2869][2] and [issue 4192][3] - collectively Crossplane's most upvoted issue of
+all time. A few years ago we broke up the largest providers into [families of
+smaller providers][4]. This helped alleviate the issue, but some large
+Crossplane deployments still work around it - for example by building their own
+providers that have only the MRs they need.
+
+## Proposal
+
+I propose we introduce a new type - ManagedResourceDefinition (MRD).
+
+An MRD would be a lightweight abstraction on a CRD:
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: ManagedResourceDefinition
+metadata:
+  name: nopresources.nop.crossplane.io
+spec:
+  group: nop.crossplane.io
+  names:
+    categories:
+    - nop
+    kind: NopResource
+    listKind: NopResourceList
+    plural: nopresources
+    singular: nopresource
+  scope: Cluster
+  versions:
+    name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        # Omitted for brevity
+    served: true
+    storage: true
+  connectionDetails:
+  - name: password
+    description: Definitely real password for this resource that does nothing
+  desiredState: Active
+```
+
+An MRD would be schematically identical to a CRD, but have two additional
+`spec` fields:
+
+* `spec.connectionDetails` - An array of connection detail keys and descriptions
+* `spec.desiredState` - Toggles whether the underlying CRD is created or not
+
+I propose we update all providers to deliver MRDs - not CRDs - as their package
+payload.
+
+I propose all MRDs be inactive (`spec.desiredState: Inactive`) by default, and
+that we add another new type to automatically activate matching MRDs. This new
+type would be called a ManagedResourceActivationPolicy (MRAP).
+
+Here's an example MRAP:
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: ManagedResourceActivationPolicy
+metadata:
+  name: aws
+spec:
+  activate:
```

Comment:
What if this was part of the provider spec and the package manager can selectively install and remove the CRDs in the package?